ros2_ws/src/otto_deliberation/include/otto_deliberation/nav2_action_client.h

```diff
@@ -18,11 +17,45 @@ class Nav2ActionClient
     void SendGoal(const NavPath& targetPath, ResultCallback resultCallback, bool goBlind = false, bool reverse = false, bool blindHighSpeed = false);
 
 private:
+    template <typename ActionType>
+    void SendSpecializedGoal(const typename ActionType::Goal& goal, ResultCallback callback, typename rclcpp_action::Client<ActionType>::SharedPtr m_client)
+    {
+        using GoalHandleType = rclcpp_action::ClientGoalHandle<ActionType>;
+        auto sendGoalOptions = typename rclcpp_action::Client<ActionType>::SendGoalOptions();
+        sendGoalOptions.goal_response_callback = [&logger = m_actionLogger, callback](std::shared_ptr<GoalHandleType> future)
+        {
+            auto goalHandle = future.get();
+            if (!goalHandle)
+            {
+                RCLCPP_ERROR(logger, "Goal was rejected by server");
+                callback(false);
+            }
+            else
+            {
+                RCLCPP_INFO(logger, "Goal accepted by server, waiting for result");
+            }
+        };
+        sendGoalOptions.feedback_callback = [](typename GoalHandleType::SharedPtr, const std::shared_ptr<const typename GoalHandleType::Feedback>)
+        {
+        };
+        sendGoalOptions.result_callback = [&logger = m_actionLogger, callback](const typename GoalHandleType::WrappedResult& result)
+        {
+            switch (result.code)
```

Comment:
Do we need a switch with one case only?

Reply:
I initially left this, leaving some space for handling other result codes. However, you have a point - I changed this.